### ISSUE TITLE
Gulp4 for new prod apps?

### ISSUE THREAD

**Issue by demisx** [gulpjs/gulp#824]
We are starting a new project that's targeting live date in Q1 2015. I am thinking maybe this is a good opportunity to try out G4. Do you think it's ready? Do current gulp 3 plugins need to explicitly support v4 or will they work as is? Thanks.

**Comment by dmackerman** [comment]
Thanks @demisx! We're starting a new project and looking to use Gulp 4 as well. Going to look at migrating what we currently have.

**Comment by demisx** [comment]
Wish you luck on your migration journey. :) I've blogged with my installation steps: http://demisx.github.io/gulp4/2015/01/15/install-gulp4.html

**Comment by johnpapa** [comment]
I installed it like this:


```
npm install --save-dev git://github.com/gulpjs/gulp.git#4.0

npm install –g git://github.com/gulpjs/gulp-cli.git#4.0
```

**Comment by johnpapa** [comment]
The migration is super easy. There is a quick path and a more elegant one. The quick path is is simply converting the task definitions. The more elegant one takes advantage of the new features more and I tended to restructure my task definitions and dependencies a bit to get the rewards. But .... as I said ... migration is super easy

**Comment by johnpapa** [comment]
Here is a gulp4 conversion i recently did. The entire project is here https://github.com/johnpapa/gulp-patterns/tree/gulp4  and the gulpfile itself is here https://github.com/johnpapa/gulp-patterns/blob/gulp4/gulpfile.js

**Comment by demisx** [comment]
True. Besides rewriting tasks to the new format and replacing `run-sequence` with `gulp.series` and `gulp.parallel`, I don't recall doing much else.

**Comment by johnpapa** [comment]
I only found one plugin that failed. gulp-task-listing ... for obvious reason :)

**Comment by demisx** [comment]
@johnpapa Would recommend to install gulp-cli globally and full gulp locally as a standard Gulp 4 install?

**Comment by johnpapa** [comment]
you need either gulp or gulp-cli globally AFAIK

and you of course need gulp locally

**Comment by johnpapa** [comment]
I recorded a 4.5 hour course on Gulp that will be published on www.Pluralsight.com in a few weeks. I also included a section on how to migrate to Gulp 4 using the alpha/pre-release.  I'm recording a First Look at Gulp 4 too, which I'll release in Feb/Mar assuming no major changes happen in Gulp4 :)

**Comment by stringparser** [comment]
@johnpapa could you let us know once is there?

**Comment by johnpapa** [comment]
Certainly.

**Comment by dmackerman** [comment]
Just migrated our gulpfile over to 4 as well, and I can echo the sentiments of other here that refactoring our `runSequence` blocks into `gulp.series` and `gulp.parallel` was really all it took.

**Comment by hydrosquall** [comment]
Confirming that @johnpapa 's course has been published and the migration video has been posted- you can find them [here](http://www.pluralsight.com/courses/javascript-build-automation-gulpjs). (It's an excellent course btw).

**Comment by stringparser** [comment]
@hydrosquall thanks man

**Comment by hydrosquall** [comment]
@johnpapa one caveat with the tutorial- when I installed gulp 4 alpha locally, an important plugin broke. Below is the relevant error message.

    npm WARN unmet dependency C:\Users\Cameron\gulp-dev\node_modules\gulp-nodemon requires gulp@'^3.8.11' but will load
    npm WARN unmet dependency C:\Users\Cameron\gulp-dev\node_modules\gulp,
    npm WARN unmet dependency which is version 4.0.0-alpha.1

The gulp task "test" works but none of the tasks that involve nodemon are working, so I'll have to drop back to Gulp 3 and undo all the changes to the gulpfile.

I made a note on the nodemon issues page: https://github.com/JacksonGariety/gulp-nodemon/issues/46#issuecomment-124252642
